model/IntOrObject.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 The Serverless Workflow Specification Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
+// +k8s:openapi-gen=true
+
+type IntOrObject struct {
+	IntVal *int
+	ObjVal Object
+}
+
+type Object interface {
+	DeepCopyObject() Object
+}
+
+type custom struct {
+	Object interface{}
+}
+
+func (o custom) DeepCopyObject() Object {
+	return o
+}
+
+type MyString struct {
+	Value string
+}
+
+type MyInt struct {
+	Value int
+}
+
+func (m MyInt) DeepCopyObject() Object {
+	return m
+}
+
+func (m MyString) DeepCopyObject() Object {
+	return m
+}
+
+func (intObj *IntOrObject) UnmarshalJSON(data []byte) error {
+
+	var test interface{}
+	err := json.Unmarshal(data, &test)
+	if err != nil {
+
+		return err
+	}
+	switch val := test.(type) {
+	case string:
+		var a MyString
+		err = json.Unmarshal(data, &a.Value)
+		if err != nil {
+			fmt.Println(err)
```

Comment:
should remove the fmt.Println

Reply:
it was there for debugging purposes, removed